code/game/turfs/simulated/floor/misc_floor.dm

```diff
@@ -100,6 +100,11 @@
 	var/image/overlay_image = image('icons/misc/beach.dmi', icon_state = "water5", layer = ABOVE_MOB_LAYER)
 	overlay_image.plane = GAME_PLANE
 	overlays += overlay_image
+	RegisterSignal(src, COMSIG_ATOM_INITIALIZED_ON, .proc/InitializedOn)
+
+/turf/simulated/floor/beach/water/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_ATOM_INITIALIZED_ON)
```

Comment:
Do I even need to do this here and on the other turf?

Reply:
Nope, `datum/Destroy` automatically handles all signal unregistering.